README.md

```diff
@@ -78,37 +79,61 @@ You might not have to sort the output, if you run through reads
 
-### Problems encountered if any. 
+### Problems encountered if any.
 
-*Describe this here.*
+Our own testing revealed a flaw in constructing border arrays, meaning that in some cases, when two matches overlapped in the string by a single character, we would not recognize the second match. To add insult to injury, this was already the second, improved version of calculating the border arrays, after a way too simple approach on the first try.
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+In order to verify that both the naïve and the linear algorithm work as intended, we decided to generate a test with the solutions that tries both regular and extreme situations. The test tries the following cases:
 
-### Experiments validating the running time.
+- Short patterns and genomes (chains), with length going from 0 to 10 characters (selected randomly)
+- Random chains
+- Random chains, having a chance of repeating the previous character
+- Random chains, having a chance of having a long sequence made up by a single character
+- Random chains, where a character cannot appear consecutively two or more times.
+- Fibonacci chains, made up by two different characters being concatenated following a Fibonacci distribution
+
+All of these genomes (except the short ones) have a length between 2.000 and 10.000 characters, whilst the patters length varies between 3 and 200. Moreover, for each pair of chains the genome is adapted, forcing to include the pattern a random number of times between 0 and 10 (if the randomness have made up more). The inclusion of this patterns has a high probability of having four extreme situations:
 
-For this section, you should address the following:
+- Pattern matching at the beginning of the genome
+- Pattern matching at the end of the genoms
+- Two pattern matchings next to each other
+- Two pattern matchings overlapping one with each other
 
-* An experiment that verifies that your implementation of `naive` uses no more time than O(nm) to find all occurrences of a given pattern in a text. Remember to explain your choice of test data. What are “best” and “worst” case inputs? 
+### Experiments validating the running time.
 
-* An experiment that verifies that your implementations of `lin` use no more time than O(n+m) to find all occurrences of a given pattern in a text. Remember to explain your choice of test data. What are “best” and “worst” case inputs?
+For both implementations of the naive and linear algorithm a polynomial regression was fitted to the runtime experiments results. For each graph only one variable (string being searched upon and pattern being searched for) length increased while the other remained the same, with the exception of the graphs where both increase at the same time. All the graphs were plotted two for each algorithm type, with each time the difference being worst or best case input.
```

Comment:
Regression is rarely a useful thing here, but okay. What did the regression tell you? Did you do an ANOVA or something to test for an interaction term, or how did you approach it?